examples/gadgets/simple/trace_dns/main.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	igjson "github.com/inspektor-gadget/inspektor-gadget/pkg/datasource/formatters/json"
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager"
+	ocihandler "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/simple"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/socketenricher"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+)
+
+func do() error {
+	const opPriority = 50000
```

Comment:
How do people know the priority they need to set so that their operator is called after the other operators?

Reply:
Good question. What I did was to check the priority used by other operators.

Currently it's the operator that tells that's the priority for the `Subscribe`, it requires operators to know the priorities of other operators (the CLI needs to be sure it has higher priority than the localmanager, formatter, etc). I'm really wondering if this is the right approach, perhaps we can move this responsibility to the caller? It could have a way to define the order of the operators and it should be sure they're called in the right order satisfying the dependencies. 

I know @flyth put a lot of thinking into this. Do you have strong opinions for one or another approach?

Reply:
As #2709 introduces the difference between Packet and Data (with it's own priority handling), I suggest we define only three priorities as constants for now:

`PriorityDefault` as 0, `PriorityHigh` as something like -50000 and `PriorityLow` as 50000 (or -PriorityHigh). If a dev/user needs something in between, this allows a lot of room. But for normal cases the default priority should work fine. Same can be used for Packets, as they're always handled after the Data subscriptions.